honeybadger-java-shaded/pom.xml

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>honeybadger-parent</artifactId>
+        <groupId>io.honeybadger</groupId>
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>honeybadger-java-shaded</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.honeybadger</groupId>
+            <artifactId>honeybadger-java</artifactId>
+            <version>2.0.0-SNAPSHOT</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
+            <version>${dependency.slfj.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>${maven-shade-plugin.version}</version>
+                <configuration>
+                    <minimizeJar>false</minimizeJar>
+                    <createDependencyReducedPom>true</createDependencyReducedPom>
+                    <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+                    <artifactSet>
+                        <excludes>
+                            <exclude>com.google.code.findbugs:jsr305</exclude>
+                            <exclude>org.slf4j:slf4j-api</exclude>
+                        </excludes>
+                    </artifactSet>
+                    <relocations>
+                        <relocation>
+                            <pattern>com.google</pattern>
+                            <shadedPattern>io.honeybadger.com.google</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>org.apache</pattern>
+                            <shadedPattern>io.honeybadger.org.apache</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.jcabi</pattern>
+                            <shadedPattern>io.honeybadger.com.jcabi</shadedPattern>
+                        </relocation>
+                        <relocation>
+                            <pattern>com.github</pattern>
+                            <shadedPattern>io.honeybadger.com.github</shadedPattern>
+                        </relocation>
+                    </relocations>
```

Comment:
There was no relocation added for the jackson dependency, so the jackson libraries would conflict with anyone's existing libraries who imported the shaded jar.

I would add:
```xml
                        <relocation>
                            <pattern>com.fasterxml</pattern>
                            <shadedPattern>io.honeybadger.com.fasterxml</shadedPattern>
                        </relocation>
```

Reply:
Will push that change in a bit.

Reply:
I believe this is working with the shaded artifact. I thought I was seeing an issue, but it turned out that the servlet app in crywolf explicitly sets the honeybadger.api_key in its web.xml, overriding my environment variable. I think I noticed that the first time I used crywolf, but I missed it today. Not sure if I should change CryWolf instructions or remove the key from the pom. (Or change the precedence of configuration methods?)